CMakeLists.txt

```diff
@@ -185,9 +185,26 @@ if("${MD_SIM}" STREQUAL "LS1_MARDYN")
       message(FATAL_ERROR "xerces-c lib not found. Disable VTK support, if you do not need it.")
     endif()
   endif()
+  if(BUILD_WITH_ADIOS2 AND NOT LS1_CACHE_ENABLE_ADIOS2)
+    message(FATAL_ERROR "Please rebuild ls1 with ENABLE_ADIOS2 turned on, if adios2 is needed, else turn BUILD_WITH_ADIOS2 off")
+  endif()
   if(LS1_CACHE_ENABLE_ADIOS2)
-    enable_language(C)
-    find_package(ADIOS2 REQUIRED)
+    if(NOT BUILD_WITH_ADIOS2)
+      message(WARNING "ls1 built with adios2, but BUILD_WITH_ADIOS2 not enabled by MaMiCo!\nProceeding to enable adios2 only for ls1")
+    endif()
+    if(LS1_CACHE_FIND_PACKAGE_ADIOS2)
+      enable_language(C)
+      find_package(ADIOS2 REQUIRED)
+      message(STATUS "Using preinstalled adios2 for ls1")
+    else()
+      enable_language(C)
+      find_package(ADIOS2 REQUIRED PATHS ${LS1_SRC_DIR}/${LS1_BUILD_DIR_NAME}/_deps/adios2fetch-build NO_DEFAULT_PATH)
+      if(BUILD_WITH_ADIOS2)
+        message(WARNING "Using bundled adios2 with ls1 for MaMiCo, overwriting previous package")
```

Comment:
The previous package is not always overwritten!  
Consider the following scenario:
Adios version fetched by ls1: “2.7.1.436”
Adios version installed using spack: “2.8.0”
The project was already configured with the spack version previously (e.g. using SimpleMD and adios2).

Now, switch the MD solver to ls1 built with adios2.  
In this case, CMake will not overwrite the spack version with the fetched version.

I did not find a way to clear/unset/unfind the adios package prior to calling `find_package(ADIOS2 ... NO_DEFAULT_PATH)` and I am not sure how big this issue is.  
Can there be compatibility issues if two different adios versions are used for ls1 and MaMiCo?

I would replace these lines with the following:
```CMake
message(STATUS "Using bundled adios2 for ls1")
if(BUILD_WITH_ADIOS2)
        message(WARNING "Both ls1 and MaMiCo will use adios2. Double check the library versions/paths!")
endif()

```

Reply:
Due to line 111, no overwriting actually happens, so that's misleading I guess. I'll still use the lines you've given.